luaui/Include/blueprint_substitution/logic.lua

```diff
@@ -0,0 +1,356 @@
+        if primarySourceSide then
+            sourceSideInfo = primarySourceSide
+            if numSourceSides > 1 then
+                sourceSideInfo = sourceSideInfo .. " (Mixed)" 
+            end
+        else
+            sourceSideInfo = "Unknown Side Buildings"
+        end
+    end
+
+    return {
+        unitCount = #blueprint.units,
+        buildingUnitCount = buildingUnitCount,
+        primarySourceSide = primarySourceSide,
+        numSourceSides = numSourceSides,
+        sourceSideInfo = sourceSideInfo,
+        sideCounts = sideCounts
+    }
+end
+
+function BlueprintSubLogic.processBlueprintSubstitution(originalBlueprint, targetSide)
+    Spring.Log("BlueprintSubLogic", LOG.DEBUG, string.format("Starting substitution for BUILDINGS ONLY. Source Side: %s, Target Side: %s, Units: %d",
+                                 originalBlueprint.sourceInfo.primarySourceSide,
+                                 targetSide,
+                                 #originalBlueprint.units))
+
+    local enrichedUnits = table.map(originalBlueprint.units, function(unit)
+        local originalUnitDefID = unit.unitDefID
+        local originalUnitDef = UnitDefs[originalUnitDefID]
+        local unitNameLower = originalUnitDef and originalUnitDef.name:lower() or ("unknown_defid_" .. originalUnitDefID)
+        
+        local record = {
+            blueprintUnitID = unit.blueprintUnitID,
+            originalUnitDefID = originalUnitDefID,
+            position = unit.position,
+            facing = unit.facing,
+            originalUnitName = unitNameLower,
+            status = "unknown", 
+            newUnitDefID = originalUnitDefID,
+            translatedHumanName = "N/A",
+            sourceSide = nil,
+            equivalentUnitName = nil
+        }
+
+        local buildingData = BlueprintSubLogic.MasterBuildingData[unitNameLower]
+
+        if buildingData then
+            record.sourceSide = buildingData.side
+            record.translatedHumanName = buildingData.translatedHumanName
+
+            if record.sourceSide and record.sourceSide ~= targetSide then
+                record.equivalentUnitName = buildingData.equivalents[targetSide]
+
+                if record.equivalentUnitName then
+                    local foundDefID = unitNameToDefIDMap[record.equivalentUnitName]
+                    if foundDefID then
+                        record.status = "success"
+                        record.newUnitDefID = foundDefID
+                    else
+                        record.status = "failed_lookup"
+                    end
+                else
+                    record.status = "failed_no_mapping"
+                end
+            else
+                record.status = "skipped_same_side" 
+            end
+        else
+            record.status = "skipped_not_building"
+        end
+        
+        -- Use LOG.DEBUG directly
+        Spring.Log("BlueprintSubLogic", LOG.DEBUG, string.format("Mapped unit %s: Status=%s", 
```

Comment:
Comment not precise enough
```suggestion
        -- Use LOG.DEBUG directly and also call string.format with two %s format specifiers
        Spring.Log("BlueprintSubLogic", LOG.DEBUG, string.format("Mapped unit %s: Status=%s", 
```

Reply:
Is it ok if I just delete it? That suggested change feels ironically mocking but I honestly can't tell. That comment was gemini 2.5 being derpy, verbose, and disobedient.